be/src/exec/spill/log_block_manager.cpp

```diff
@@ -0,0 +1,290 @@
+}
+
+class LogBlock : public Block {
+public:
+    LogBlock(LogBlockContainerPtr container, size_t offset) : _container(container), _offset(offset) {}
+
+    virtual ~LogBlock() {
+        if (_reader != nullptr) {
+            _reader.reset();
+        }
+    }
+
+    size_t offset() const { return _offset; }
+    size_t length() const { return _length; }
+
+    LogBlockContainerPtr container() const { return _container; }
+
+    Status append(const std::vector<Slice>& data) override {
+        size_t total_size = 0;
+        std::for_each(data.begin(), data.end(), [&](const Slice& slice) { total_size += slice.size; });
+        RETURN_IF_ERROR(_container->ensure_preallocate(total_size));
+        RETURN_IF_ERROR(_container->append_data(data));
+        _length += total_size;
+        return Status::OK();
+    }
+
+    Status flush() override { return _container->flush(); }
+
+    Status read_fully(void* data, int64_t count) override {
+        if (_reader == nullptr) {
+            ASSIGN_OR_RETURN(_reader, _container->get_block_reader(_offset, _length));
+        }
+        return _reader->read_fully(data, count);
+    }
+
+    std::string debug_string() override {
+        return fmt::format("LogBlock[container={}, offset={}, len={}]", _container->path(), _offset, _length);
+    }
+
+private:
+    LogBlockContainerPtr _container;
+    size_t _offset;
+    size_t _length = 0;
+    std::unique_ptr<LogBlockReader> _reader;
+};
+
+LogBlockManager::~LogBlockManager() {
+    for (auto& container : _full_containers) {
+        container.reset();
+    }
+    for (auto& [dir, container_map] : _available_containers) {
+        for (auto& [_, containers] : *container_map) {
+            containers.reset();
+        }
+        std::string container_dir = dir->dir() + "/" + print_id(_query_id);
+        WARN_IF_ERROR(dir->fs()->delete_dir(container_dir),
+                      fmt::format("cannot delete spill container dir: {}", container_dir));
+    }
+}
+
+Status LogBlockManager::open() {
+    return Status::OK();
+}
+
+void LogBlockManager::close() {}
+
+StatusOr<BlockPtr> LogBlockManager::acquire_block(const AcquireBlockOptions& opts) {
+    AcquireDirOptions acquire_dir_opts;
+#ifdef BE_TEST
+    ASSIGN_OR_RETURN(auto dir, _dir_mgr->acquire_writable_dir(acquire_dir_opts));
+#else
+    ASSIGN_OR_RETURN(auto dir, ExecEnv::GetInstance()->spill_dir_mgr()->acquire_writable_dir(acquire_dir_opts));
+#endif
+
+    ASSIGN_OR_RETURN(auto block_container, get_or_create_container(dir, opts.plan_node_id, opts.name));
+    auto block = std::make_shared<LogBlock>(block_container, block_container->size());
+    return block;
+}
+
+Status LogBlockManager::release_block(const BlockPtr& block) {
+    auto log_block = down_cast<LogBlock*>(block.get());
+    auto container = log_block->container();
+    TRACE_SPILL_LOG << "release block: " << block->debug_string();
+    bool is_full = container->size() >= config::experimental_spill_max_log_block_container_bytes;
```

Comment:
check `experimental_spill_max_log_block_container_bytes` whether is valid or not?

Reply:
> check `experimental_spill_max_log_block_container_bytes` whether is valid or not?

fixed